src/main/java/startsteps/ECommerceShop/service/CartService.java

```diff
@@ -133,4 +135,17 @@ public CartResponse removeProduct(Long productId, User user) {
             throw new ProductNotFoundException("Product with ID " + productId + " not found in your cart.");
         }
     }
+
+    @Transactional
+    public void clearCart(User user) {
+        try{
+            Cart cart = user.getCart();
+        if(cart!=null && !cart.getCartProductList().isEmpty()) {
+            cart.getCartProductList().clear();
+            updateCartTotalPrice(cart);
+            cartRepository.save(cart);
+        }
+        }catch (Exception exception){
+            exception.printStackTrace();}
```

Comment:
not sure we just want to print stack trace. either log it or react on it and do something with exception

Reply:
I think i dont need it, I added try-catch block, because I was searching issue: why products in cart don't removing after placing order, issue was  fixed, but forget to remove the try-catch block